bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/network/SiemensHvacConnectorImpl.java

```
@@ -0,0 +1,489 @@
+            logger.debug("siemensHvac:doAuth:error() {}", ex.getLocalizedMessage());
+        } finally {
+        }
+    }
+
+    public @Nullable String DoBasicRequest(String uri) throws Exception {
+        return DoBasicRequest(uri, null);
+    }
+
+    public @Nullable String DoBasicRequestAsync(String uri, @Nullable SiemensHvacCallback callback) throws Exception {
+        return DoBasicRequest(uri, callback);
+    }
+
+    public @Nullable String DoBasicRequest(String uri, @Nullable SiemensHvacCallback callback) throws Exception {
+        if (sessionIdHttp == null) {
+            doAuth(true);
+        }
+
+        if (sessionId == null) {
+            doAuth(false);
+        }
+
+        try {
+            String baseUri = baseUrl;
+
+            String mUri = uri;
+            if (!mUri.endsWith("?")) {
+                mUri = mUri + "&";
+            }
+            if (mUri.indexOf("main.app") >= 0) {
+                mUri = mUri + "SessionId=" + sessionIdHttp;
+            } else {
+                mUri = mUri + "SessionId=" + sessionId;
+            }
+
+            logger.debug("Execute request: {}", uri);
+            CookieStore c = httpClient.getCookieStore();
+            java.net.HttpCookie cookie = new HttpCookie("SessionId", sessionIdHttp);
+            cookie.setPath("/");
+            cookie.setVersion(0);
+
+            c.add(new URI(baseUri), cookie);
+
+            logger.debug("Execute request: {}", uri);
+            final Request request = httpClient.newRequest(baseUri + mUri);
+            request.method(HttpMethod.GET);
+
+            ContentResponse response = executeRequest(request, callback);
+            if (callback == null && response != null) {
+                int statusCode = response.getStatus();
+
+                if (statusCode == HttpStatus.OK_200) {
+                    String result = response.getContentAsString();
+
+                    return result;
+                }
+            }
+        } catch (Exception ex) {
```

Do you need to catch all exceptions here?